jdaviz/core/template_mixin.py

```diff
@@ -3336,15 +3347,27 @@ def selected_obj(self):
         # from the data collection
         return self.get_object(cls=self.default_data_cls)
 
-    def selected_spectrum_for_spatial_subset(self,
-                                             spatial_subset=SPATIAL_DEFAULT_TEXT,
-                                             use_display_units=True):
-        if spatial_subset == SPATIAL_DEFAULT_TEXT:
-            spatial_subset = None
+    def get_selected_spectrum(self, use_display_units=True):
+        # retrieves the 1d spectrum
+        if len(self.selected_obj.shape) == 3:
+            # then this is a cube, but we want the 1D spectrum,
+            # so we can pass through the Spectral Extraction plugin
+            if self.plugin.config != 'cubeviz':
+                raise ValueError("extracting a spectrum from a cube only supported in cubeviz")
+            # we need to get the 1d extracted spectrum for the cube
+            spec_extract = self.app._jdaviz_helper.plugins['Spectral Extraction']._obj
+            sp = spec_extract._extract_in_new_instance(self.selected,
+                                                       function=self._spectral_extraction_function,
+                                                       add_data=False)
+            return self.plugin._specviz_helper._handle_display_units(sp, use_display_units)
         return self.plugin._specviz_helper.get_data(data_label=self.selected,
-                                                    spatial_subset=spatial_subset,
                                                     use_display_units=use_display_units)
 
+    @cached_property
+    def selected_spectrum(self):
+        # TODO: do we need to reset this cache on a change to display units?
```

Comment:
Probably?

Reply:
added - although we won't have a good test case for this until #2781 is also in.  But the logic is simple enough so might as well have it in now and be future-proofed (hopefully).